Check Windows files for sanity as well

gilles-peskine-arm opened the pull request:
Consistently use Windows line endings in Windows files. Include Windows files in `check-files.py`.

bensze01 commented on tests/scripts/check-files.py:
```diff
@@ -182,17 +182,17 @@ def __init__(self, log_file):
-            "/Makefile.inc",
+            ".sh",
             "/CMakeLists.txt",
             "/ChangeLog"
```
These two lines should be swapped.

gilles-peskine-arm replied:
No, the order is right. But the `ChangeLog` line was missing a comma! Fixed in rebase.

bensze01 replied:
Yeah, you're right.

bensze01 commented on tests/scripts/check-files.py:
```diff
@@ -180,9 +180,9 @@ def __init__(self, log_file):
             ".c", ".h", ".sh", ".pl", ".py", ".md", ".function", ".data",
-            "Makefile", "Makefile.inc", "CMakeLists.txt", "ChangeLog"
+            "/Makefile", "/Makefile.inc", "/CMakeLists.txt", "/ChangeLog"
+        self.extensions_to_check = (
```
This line should go above the strings.

gilles-peskine-arm replied:
Indeed, this was a bad partial commit. I rewrote it.

bensze01 commented on tests/scripts/check-files.py:
```diff
@@ -181,10 +181,23 @@ def __init__(self, log_file):
+            "/Makefile",
+            "/Makefile.inc",
+            "/CMakeLists.txt",
+            "/ChangeLog",
```
`ChangeLog `shouldn't have a comma after it in *this* commit, since here it is the last element of the tuple. It only needs a comma in the next commit.

gilles-peskine-arm replied:
Yes, it should have a comma in this commit. Having a comma after the last element of a multiline list is recommended, precisely in case the element ends up not being the last one anymore.

bensze01 replied:
Ok.

mpg commented on scripts/data_files/vs2010-app-template.vcxproj:
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
You're also removing the BOM (U+FEFF), here and in every file touched, which is not mentioned in the commit message. Is that intentional? Will VS accept those files without a BOM?

gilles-peskine-arm replied:
Well-spotted! No, the change wasn't intentional. I don't know if the BOM is required, and since I don't know I don't want to change anything. I'll fix it in a rebase and pay closer attention to what my editor is doing.

gilles-peskine-arm replied:
Looks like Emacs automatically removes a leading U+FEFF at the beginning of an XML file encoded in UTF-8. [It shouldn't be there](https://www.opentag.com/xfaq_enc.htm#enc_bom), but if Visual Studio prefers technically invalid XML, I'll keep the XML technically invalid.

mpg replied:
Thanks for fixing! `git show --patch -w` now shows an empty diff as it should

mpg commented on tests/scripts/check-files.py:
```diff
@@ -113,16 +119,21 @@ class Utf8BomIssueTracker(FileIssueTracker):
 
     heading = "UTF-8 BOM present:"
 
+    files_exemptions = frozenset([".vcxproj", ".sln"])
```
Oh, this looks like the removal of the BOM in the previous commit wasn't quite intentional, then.

gilles-peskine-arm commented:
I rewrote the last commit "Systematically use Windows line endings in Windows files" to not remove the leading U+FFFE.

I also added two commits before that to check that Windows files don't contain a CR without a LF, and to fix a file where this happened.

mpg commented:
The tests are all passing except the Mbed OS test that are not expected to pass now, and an instance of the "reconnect" test in `ssl-opt.sh` which is a known issue with a fix pending in #3132 - so everything's fine as far as this PR is concerned.

mpg commented:
Note: the CI's pr-merge job passed except mbed-os, and the pr-head job passed except the mbed-os test and an instance of a 'hard reconnect' test failing in `ssl-opt.sh`, which is a known issue that has been fixed by #3132 in the meantime.

gilles-peskine-arm commented:
Backports are up: #3169 #3170
